iotfunctions/anomaly.py

```diff
@@ -507,10 +509,18 @@ def prepare_data(self, dfEntity):
     def execute(self, df):
 
         df_copy = df.copy()
-        entities = np.unique(df_copy.index.levels[0])
-        logger.debug(str(entities))
+        unique_entities = np.unique(df_copy.index.levels[0])
+        entities = []
```

Comment:
Maybe you should consider something like as it's shorter and (probably) faster
```
entities = np.unique(df_copy.index.levels[0]),
if self.entity_list is not None and isinstance(self.entity_list, np. ndarray):
        entities = np.intersect1d(entities, self.entity_list)
```

Reply:
That's neat.
I learned a new trick